storage/rocksdb_storage.cpp

```diff
@@ -385,3 +385,74 @@ rocksdb_storage::load_snapshot(const std::string& data)
 
     return false;
 }
+
+
+void
+rocksdb_storage::remove_range(const bzn::uuid_t& uuid, const std::string& first, const std::string& last)
+{
+    std::shared_lock<std::shared_mutex> lock(this->lock); // lock for read access
```

Comment:
you want write access which means a std::lock_guard should be used

Reply:
Thanks. I didn't grok the lock semantics. Will fix.